src/components/TodoForm.js

```diff
@@ -1,48 +1,76 @@
-import React, { Component } from "react";
+import React, { useReducer, useState } from "react";
 import { Form, FormGroup, Input, Button } from "reactstrap";
 
-class TodoForm extends Component {
-  constructor() {
-    super();
-    this.state = { inputVal: "" };
+const todoReducer = (state, action) => {
+  switch (action.type) {
+    case "ADD_TODO":
+      return {
+        ...state,
+      };
   }
+};
 
-  handleSubmit = (e) => {
+const TodoForm = (props) => {
+  const [inputVal, setInputVal] = useState("");
+  const [todos, setTodos] = useState(props.todos);
+
+  const addTodo = (val) => {
+    const newTodos = [
+      ...todos,
+      {
+        task: val,
+        id: Date.now(),
+        completed: false,
+      },
+    ];
+    setTodos(newTodos);
+  };
+
+  const removeCompleted = () => {
+    const filteredTodos = todos.filter((todo) => {
+      if (todo.completed) {
+        return false;
+      }
+      return true;
+    });
+    setTodos(filteredTodos);
+  };
+
+  const handleSubmit = (e) => {
     e.preventDefault();
-    if (this.state.inputVal === "") return;
+    if (inputVal === "") return;
     console.log("Sunt in functia handleSubmit()");
-    this.props.addTodo(this.state.inputVal);
-    this.setState({ inputVal: "" });
+    addTodo(inputVal);
+    setInputVal({ inputVal: "" });
```

Comment:
iata aici e problema. Daca te uiti atent, in componenta de clasa la tine state-ul era obiect asai? respectiv state-ul pentru input arata in felul urmator: `{ inputValue: "bla bla" }`. Dar in componenta function la tine state-ul pentru input e un string(linia 14)

Reply:
Ok, am inteles, dar totusi nu-mi dau seama cum sa-l scriu aici acest state sa fie corect

Reply:
pai cum faceam noi inainte cu controlled forms si useState

Reply:
setInputVal(inputVal: "" );

Reply:
fara acolade

Reply:
inputValue ce tip este? iar setterul sau trebuie sa fie apelat tot cu acelasi tip de care este si inputValue, nu?

Reply:
astfel e destul de simplu, tu vrei ca `inputVal` sa fie gol dupa ce faci submit. Iar `setInputVal` doar ca setter seteaza valoarea celeilalte variabile cu orice nu i-ai da. astfel raspunsul e `setInputVal('')`

Reply:
este string, iar eu il setez ca obiect

Reply:
pai de ce sa il setezi ca obiect? 😄

Reply:
> astfel e destul de simplu, tu vrei ca `inputVal` sa fie gol dupa ce faci submit. Iar `setInputVal` doar ca setter seteaza valoarea celeilalte variabile cu orice nu i-ai da. astfel raspunsul e `setInputVal('')`

Bine, multumesc